src/test/unit_tests/test_postgres_upgrade.js

```diff
@@ -0,0 +1,250 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+// setup coretest first to prepare the env
+// const coretest = require('./coretest');
+const mocha = require('mocha');
+const { Migrator } = require('../../upgrade/migrator');
+const assert = require('assert');
+const EventEmitter = require('events').EventEmitter;
+const _ = require('lodash');
+const postgres_client = require('../../util/postgres_client');
+const wtf = require('wtfnode');
+
+
+// coretest.setup({ pools_to_create: [], db_name: 'upgradetest' });
+process.env.CORETEST = 'upgrade-test';
+const COLLECTION_SIZE = 3000;
+// const COLLECTION_NUMBER = 100;
+const default_schema = {
+    id: 'default_schema',
+    type: 'object',
+    required: [
+        '_id',
+        'key1',
+        'key2',
+        'key3',
+    ],
+    properties: {
+        _id: { objectid: true },
+        key1: { type: 'string' },
+        key2: { type: 'string' },
+        key3: { type: 'string' },
+    }
+};
+
+class TestCollection {
+
+    /**
+     * @param {TestCollection} col
+     * @returns {nb.DBCollection}
+     */
+    static implements_interface(col) { return col; }
+
+    constructor(col, size) {
+        TestCollection.implements_interface(this);
+        this.schema = col.schema;
+        this.name = col.name;
+        this.db_indexes = col.db_indexes;
+        this.size = size;
+        this.data = _.times(this.size, i => ({ _id: i + 1, key1: 'value1-' + i, key2: 'value2-' + i, key3: 'value3-' + i}));
+    }
+
+    async find(query, options = {}) {
+        const start = (query && query._id && query._id.$gt) || 0;
+        const end = options.limit ? start + options.limit : this.size;
+        console.log("data to be moved", this.data.slice(start, end), query);
```

Comment:
won't it make the test very chatty? I think it's better to print only on failures

Reply:
was for debugging - removed